benchmarks/array-of-coordinates.js

```diff
@@ -0,0 +1,111 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* global Buffer */
+
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var Thrift = require('../thrift').Thrift;
+var benchCasesSync = require('./lib/benchmark').benchCasesSync;
+
+var sourceFile = path.join(__dirname, 'benchmark.thrift');
+var source = fs.readFileSync(sourceFile, 'utf8');
+var thrift = new Thrift({source: source});
+
+benchCasesSync(generateMultipleCoordinatesCases([0, 1, 1e1, 1e2, 1e3, 1e4]));
+
+function generateMultipleCoordinatesCases(lengths) {
+    var cases = {};
+    for (var i = 0; i < lengths.length; i++) {
+        var length = lengths[i];
+        var subcases = generateCoordinatesCases(length);
+        var names = Object.keys(subcases);
+        for (var j = 0; j < names.length; j++) {
+            var name = names[j];
+            cases[length + ' ' + name] = subcases[name];
+        }
+    }
+    return cases;
+}
+
+function generateCoordinatesCases(count) {
+    // large array of {lat,lon}
+
+    var structuredCoordinates = [];
+    var unstructuredCoordinates = [];
+    for (var i = 0; i < count; i++) {
+        var lat = Math.random();
+        var lon = Math.random();
+        var structuredCoordinate = new thrift.Coordinate({
+            lat: lat,
+            lon: lon
+        });
+        var unstructuredCoordinate = {
+            lat: lat,
+            lon: lon
+        };
+        structuredCoordinates.push(structuredCoordinate);
+        unstructuredCoordinates.push(unstructuredCoordinate);
+    }
+
+    var structuredEnvelope = new thrift.CoordinatesHolder();
+    structuredEnvelope.coordinates = structuredCoordinates;
+
+    var unstructuredEnvelope = {
+        coordinates: unstructuredCoordinates
+    };
+
+    var lengthRes = thrift.CoordinatesHolder.rw.byteLength(structuredEnvelope);
+    var length = lengthRes.length;
+    var buffer = new Buffer(length);
+    var json = JSON.stringify(unstructuredEnvelope);
+
+    var cases = {
+        'json read': function b() {
+            JSON.parse(json);
+        },
+        'thrift read': function b() {
+            thrift.CoordinatesHolder.rw.readFrom(buffer, 0);
+        },
+        'json write pojo': function b() {
+            JSON.stringify(unstructuredEnvelope);
```

Comment:
We are calling a closure in a loop. Are we sure this is a good approach ?

I feel like having no closures makes the benchmark more reliable.